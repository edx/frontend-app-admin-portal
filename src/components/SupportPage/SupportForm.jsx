import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import { Button, Icon } from '@edx/paragon';

import RenderField from '../RenderField';
import StatusAlert from '../StatusAlert';
import TextAreaAutoSize from '../TextAreaAutoSize';

import { isRequired, isValidEmail, maxLength512 } from '../../utils';

class SupportForm extends React.Component {
  renderErrorMessage() {
    const { error: { message } } = this.props;

    return (
      <StatusAlert
        className="mt-3"
        alertType="danger"
        iconClassName="fa fa-times-circle"
        title="Unable to send your request to the edX Customer Success team."
        message={`Try refreshing your screen (${message})`}
      />
    );
  }

  renderSuccessMessage() {
    return (
      <StatusAlert
        className="mt-3"
        alertType="success"
        iconClassName="fa fa-check-circle"
        title="Request received"
        message="The edX Customer Success team will contact you soon."
        dismissible
      />
    );
  }

  render() {
    const {
      handleSubmit,
      submitting,
      invalid,
      submitSucceeded,
      submitFailed,
      error,
    } = this.props;

    return (
      <>
        {submitFailed && error && this.renderErrorMessage()}
        {submitSucceeded && this.renderSuccessMessage()}
        <div className="support-form row">
          <div className="col-12 col-md-6 col-lg-4">
            <form onSubmit={handleSubmit}>
              <Field
                name="emailAddress"
                type="email"
                component={RenderField}
                label={(
                  <>
                    Email Address
                    <span className="required">*</span>
                  </>
                )}
                validate={[isRequired, isValidEmail]}
                required
                data-hj-suppress
              />
              <Field
                name="enterpriseName"
                type="text"
                component={RenderField}
                label={(
                  <>
                    Company
                    <span className="required">*</span>
                  </>
                )}
                validate={[isRequired]}
                required
                disabled
                data-hj-suppress
              />
              <Field
                name="subject"
                type="text"
                component={RenderField}
                label={(
                  <>
                    Subject
                    <span className="required">*</span>
                  </>
                )}
                validate={[isRequired]}
                required
                data-hj-suppress
              />
              <Field
                name="notes"
                type="text"
                component={TextAreaAutoSize}
                label={(
                  <>
                    Notes
                    <span className="required">*</span>
                  </>
                )}
                validate={[isRequired, maxLength512]}
                required
                data-hj-suppress
              />
              <Button
                type="submit"
                disabled={invalid || submitting}
                className="btn-primary"
              >
                <>
                  {submitting && <Icon className="fa fa-spinner fa-spin mr-2" />}
                  Contact Support
                </>
              </Button>
            </form>
          </div>
        </div>
      </>
    );
  }
}

SupportForm.defaultProps = {
  error: null,
};

SupportForm.propTypes = {
  // props From redux-form
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  invalid: PropTypes.bool.isRequired,
  submitSucceeded: PropTypes.bool.isRequired,
  submitFailed: PropTypes.bool.isRequired,
  error: PropTypes.instanceOf(Error),

  // custom props
  match: PropTypes.shape({
    url: PropTypes.string.isRequired,
  }).isRequired,
  initialValues: PropTypes.shape({
    emailAddress: PropTypes.string.isRequired,
    enterpriseName: PropTypes.string.isRequired,
  }).isRequired,
};

export default reduxForm({ form: 'support-form' })(SupportForm);
