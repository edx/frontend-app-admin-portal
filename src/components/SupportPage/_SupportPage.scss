.support-form {
  label {
    font-weight: 600;
  }

  .required {
    color: $danger;
    margin-left: 4px;
  }

  input[type='number'] {
    width: auto;
  }

  .form-cancel-btn {
    text-decoration: underline;

    &:focus {
      box-shadow: $input-btn-focus-box-shadow;
    }
  }
}
