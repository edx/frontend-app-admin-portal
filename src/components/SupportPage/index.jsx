import React from 'react';
import { Helmet } from 'react-helmet';
import { MailtoLink } from '@edx/paragon';

import H1 from '../../components/H1';

const SupportPage = () => (
  <div className="container mt-3">
    <Helmet>
      <title>Support</title>
    </Helmet>
    <H1>Support</H1>
    <p>
      For assistance, please contact edX Enterprise Support at <MailtoLink to="customersuccess@edx.org" content=" customersuccess@edx.org" />.
    </p>
  </div>
);

export default SupportPage;
