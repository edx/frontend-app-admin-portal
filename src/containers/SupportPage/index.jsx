import { connect } from 'react-redux';

import SupportPage from '../../components/SupportPage';

const mapStateToProps = state => ({
  enterpriseName: state.portalConfiguration.enterpriseName,
  emailAddress: state.userAccount.email,
});

export default connect(mapStateToProps)(SupportPage);
