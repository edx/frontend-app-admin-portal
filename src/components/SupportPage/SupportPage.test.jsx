import React from 'react';
import renderer from 'react-test-renderer';

import SupportPage from './index';

describe('<SupportPage />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create((
        <SupportPage />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
