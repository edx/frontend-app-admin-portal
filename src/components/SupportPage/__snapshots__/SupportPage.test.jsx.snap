// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SupportPage /> renders correctly 1`] = `
Array [
  <div
    className="hero py-4 pt-md-5 pb-md-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-md-9 mb-2 mb-md-1"
        >
          <h1
            className={null}
          >
            Contact Support
          </h1>
        </div>
        <div
          className="col-12 col-md-3 text-md-right mt-2"
        >
          <img
            alt="edX logo"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container-fluid"
  >
    <div
      className="row my-3"
    >
      <div
        className="col"
      >
        <p>
          For assistance, please contact the edX Customer Success team at
           
          <a
            href="mailto:customersuccess@edx.org"
            onClick={null}
            target="_self"
          >
            customersuccess@edx.org
          </a>
          .
        </p>
      </div>
    </div>
  </div>,
]
`;
